### PR TITLE
Bring back test suite runs on GitHub actions

### PR DESCRIPTION
**Pull request by nicomiguelino**
### General

- Development will be continued on #1616.

### Running Tests Locally (Dockerized)

```bash
docker-compose -f docker-compose.test..yml up -d
docker-compose -f docker-compose.test.yml exec srly-ose-test bash bin/run_tests.sh
```

### Possible Next Steps

- Remove any unnecessary commands in `bin/run_tests.sh`.
- Ensure that all `nose` tests are passing.
- Modify the GitHub Actions workflow file so that tests could be run.
- Speed up the GitHub Actions workflow.
- Modify `README.md` and/or any relevant docs.
- If applicable, restore the command for running PhantomJS tests.
- Make local tests idempotent. Results should be the same after running it multiple times.

### CODE REVIEW

File: bin/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+prepare () {
+    mkdir -p ~/.screenly ~/.config/uzbl/ ~/screenly_assets /tmp/USB/cleanup_folder
+    # cp ansible/roles/screenly/files/uzbl-config ~/.config/uzbl/config-screenly
+    cp ansible/roles/screenly/files/screenly.conf ~/.screenly/
+    cp ansible/roles/screenly/files/screenly.db ~/.screenly/
+    cp bin/install.sh /usr/local/sbin/upgrade_screenly.sh
+    chmod 0700 /usr/local/sbin/upgrade_screenly.sh
+    echo -e "[local]\nlocalhost ansible_connection=local" > ansible/localhost
```
Comment (vpetersson) [inline bin/run_tests.sh:10]: We should be able to remove this.

File: bin/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+prepare () {
+    mkdir -p ~/.screenly ~/.config/uzbl/ ~/screenly_assets /tmp/USB/cleanup_folder
+    # cp ansible/roles/screenly/files/uzbl-config ~/.config/uzbl/config-screenly
+    cp ansible/roles/screenly/files/screenly.conf ~/.screenly/
+    cp ansible/roles/screenly/files/screenly.db ~/.screenly/
+    cp bin/install.sh /usr/local/sbin/upgrade_screenly.sh
+    chmod 0700 /usr/local/sbin/upgrade_screenly.sh
```
Comment (vpetersson) [inline bin/run_tests.sh:9]: Permission should already be set from Git.

File: bin/run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    echo -e "[local]\nlocalhost ansible_connection=local" > ansible/localhost
+    curl https://www.screenly.io/upload/ose-logo.png > /tmp/image.png
+    cp /tmp/image.png /tmp/USB/image.png
+    cp /tmp/image.png /tmp/USB/cleanup_folder/image.png
+    curl https://www.screenly.io/upload/ose-logo.png > ~/screenly_assets/image.tmp
+    curl https://www.screenly.io/upload/big_buck_bunny_720p_10mb.flv > /tmp/video.flv
+    cp tests/assets/asset.mov /tmp/asset.mov
+    export DISPLAY=:99.0
```
Comment (vpetersson) [inline bin/run_tests.sh:17]: This should only be needed from front-end tests.

**Review by vpetersson — CHANGES_REQUESTED** [review]
Looks like a good start, but it is never actually implemented in [docker-build.yaml](https://github.com/Screenly/screenly-ose/blob/master/.github/workflows/docker-build.yaml) as the first job with `needs` set to this job on the existing first step.

**Comment by nicomiguelino** [comment]
@vpetersson, as mentioned in the comments, I will continue development on #1616 instead.

**Pull request closed without merging**